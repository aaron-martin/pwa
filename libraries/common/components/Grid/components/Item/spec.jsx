/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { shallow } from 'enzyme';
import GridItem from './index';

describe('<GridItem />', () => {
  it('should render without any further props', () => {
    const wrapper = shallow(<GridItem />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should be able to render a custom tag', () => {
    const wrapper = shallow(<GridItem component="section" />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.type()).toEqual('section');
  });

  it('should add custom classes on demand', () => {
    const wrapper = shallow(<GridItem className="custom-class-name" />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.hasClass('custom-class-name')).toEqual(true);
  });
});
