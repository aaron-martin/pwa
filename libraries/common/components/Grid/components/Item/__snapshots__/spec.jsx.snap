// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridItem /> should add custom classes on demand 1`] = `
<li
  className="custom-class-name"
/>
`;

exports[`<GridItem /> should be able to render a custom tag 1`] = `
<section
  className=""
/>
`;

exports[`<GridItem /> should render without any further props 1`] = `
<li
  className=""
/>
`;
